packages/blob/README.md

```diff
@@ -129,9 +129,47 @@ async function list(options?: {
 }> {}
 ```
 
+### handleBlobUpload(options)
+
+Handles the requests to generate a client token and respond to the upload completed event. This is useful when [uploading directly from browsers](#uploading-directly-from-browsers) to circumvent the 4MB limitation of going through a Vercel-hosted route.
+
+```ts
+async function handleBlobUpload(options?: {
+  token?: string;
+  request?: IncomingMessage | Request;
+  onBeforeGenerateToken: (pathname: string) => Promise<{
+    allowedContentTypes?: string[];
+    maximumSizeInBytes?: number;
+    validUntil?: number;
+    metadata?: string;
+  }>;
+  onUploadCompleted: (body: {
+    type: 'blob.upload-completed';
+    payload: {
+      blob: BlobResult;
+      metadata?: string;
+    };
+  }) => Promise<void>;
+  body:
+    | {
+        type: 'blob.upload-completed';
+        payload: {
+          blob: BlobResult;
+          metadata?: string;
+        };
+      }
+    | {
+        type: 'blob.generate-client-token';
+        payload: { pathname: string; callbackUrl: string };
+      };
+}): string {}
+```
+
+Note: This method should be called server-side, not client-side.
+
 ### generateClientTokenFromReadWriteToken(options)
 
-Generates a single-use token that can be used from within the client. This is useful when [uploading directly from browsers](#uploading-directly-from-browsers) to circumvent the 4MB limitation of going through a Vercel-hosted route.
+Generates a single-use token that can be used from within the client. This method is called internally by `handleBlobUpload`.
```

Comment:
we may add an "Advanced use case" section later on, wdyt @vvo ?

Reply:
Yes I think that would be useful for folks willing to go full customization as we discussed (caching token, swr, ..). Can you track this in Linear? Thanks!

Reply:
https://linear.app/vercel/issue/STO-669/document-client-upload-advanced-use-case